Adjustments for configure --enable-rexgen

frank-dittrich opened the pull request:
Adjust john to "recent" rexgen changes (including some interface
changes), point out known issues and possibility that rexgen support
might have to be dropped in future releases

frank-dittrich committed:
Adjustments for configure --enable-rexgen

Adjust john to "recent" rexgen changes (including some interface
changes), point out known issues and possibility that rexgen support
might have to be dropped in future releases

solardiz commented on doc/README.librexgen:
```diff
@@ -111,9 +167,45 @@ rexgen '[0-2]password[A-C]'
+(Note the single quotes and the need to escape both '.' and '?'.)
+
+2.
+While some invalod regular expressions are correctly detected and reported,
```
You probably meant `invalid`, unless you deliberately invalidated this word. ;-)

solardiz commented on src/configure:
```diff
@@ -1395,7 +1395,7 @@ Optional Features:
                           use fork instead)
   --enable-mpi            Use OpenMPI
-  --enable-rexgen         Enable use of rexgen library (experimental)
+  --enable-rexgen         Enable use of rexgen library (experimental!)
```
Maybe replace `experimental!` with `unreliable`. This isn't really an experiment for some upcoming feature - it's a feature that's been around for a long time and never became reliable and isn't expected to become reliable now.

solardiz commented on src/configure.ac:
```diff
@@ -75,7 +75,7 @@ dnl We define "--disable-FEATURE" when default is "auto".
 AC_ARG_ENABLE([openmp-for-fast-formats], [AC_HELP_STRING([--enable-openmp-for-fast-formats], [Enable OpenMP for fast formats (usually a bad idea - use fork instead)])], [ompfast=$enableval], [ompfast=no])
 AC_ARG_ENABLE([mpi], [AC_HELP_STRING([--enable-mpi], [Use OpenMPI])], [enable_mpi=$enableval], [enable_mpi=no])
-AC_ARG_ENABLE([rexgen], [AC_HELP_STRING([--enable-rexgen], [Enable use of rexgen library (experimental)])], [enable_rexgen=$enableval], [enable_rexgen=no])
+AC_ARG_ENABLE([rexgen], [AC_HELP_STRING([--enable-rexgen], [Enable use of rexgen library (experimental!)])], [enable_rexgen=$enableval], [enable_rexgen=no])
```
Ditto.

solardiz reviewed (approved):
Looks good to me. Feel free to make the minor edits I suggested or ask for this to be merged as-is. Thank you!

solardiz commented:
Thanks, @frank-dittrich! Ready to merge now?

frank-dittrich commented:
@solardiz I fixed the misspelled word, replaced "experimental" with "unreliable" (In my initial commit I was even thinking about "b0rken"), and added another illegal regex causing a segfault.

IMHO ready to merge.
Or should I try to add workarounds in regex.c for known problematic regular expressions like empty `[]` and illegal back references? At least the `[]` should be trivial.
I don't like to add clumsy workarounds in john's regex.c, but I definitely wouldn't want to become a rexgen maintainer.
May be workarounds for broken rexgen code should be separate commits so that they can be reverted should rexgen fix these issues in the future.

frank-dittrich commented:
Just found another segfault:
```
rexgen '[äöü]'
```
Should I add this to doc/README.librexgen?

frank-dittrich commented:
@magnumripper @solardiz 
Please don't merge. I found a more stable, less troublesome rexgen version, will submit changes after I had some sleep
